server/controllers/events.js

```diff
@@ -1,61 +1,107 @@
-const Event = require("../models/Event");
+const mongoose = require("mongoose");
+const Joi = require("joi");
+const { Event } = require("../models/Event");
+const httpError = require("../utilities/httpError");
 
 module.exports = {
-  //test function
-  ping: (req, res) => {
-    return res.json({ message: "pong" });
-  },
-  create: async (req, res) => {
+  create: async (req, res, next) => {
     try {
       let data = JSON.parse(req.body.data);
-      data.forEach(obj => obj.user = req.user._id);
+      data.forEach(obj => (obj.user = req.user._id));
       await Event.insertMany(data);
       res.json({ message: "Event created!" });
     } catch (err) {
-      console.log(err);
+      next(err);
     }
   },
-  getAll: async (req, res) => {
+  getAll: async (req, res, next) => {
```

Comment:
What is the next parameter accomplishing?

Reply:
Passes the request to the next controller. i use it here mostly in the `catch` block to pass the error to the error handler